tests/integration/api/controllers/report.spec.js

```diff
@@ -0,0 +1,212 @@
+  const contact2 = utils.deepFreeze(personFactory.build({name: 'contact2', role: 'program_officer'}));
+  const placeMap = utils.deepFreeze(placeFactory.generateHierarchy());
+  const place1 = utils.deepFreeze({...placeMap.get('health_center'), contact: {_id: contact1._id}});
+  const place2 = utils.deepFreeze({...placeMap.get('district_hospital'), contact: {_id: contact2._id}});
+  const place0 = utils.deepFreeze({
+    ...placeMap.get('clinic'), contact: {_id: contact0._id}, parent: {
+      _id: place1._id, parent: {
+        _id: place2._id
+      }
+    },
+  });
+  const patient = utils.deepFreeze(personFactory.build({
+    parent: {
+      _id: place0._id, parent: {
+        _id: place1._id, parent: {
+          _id: place2._id
+        }
+      },
+    }, phone: '1234567890', role: 'patient', short_name: 'Mary'
+  }));
+  const report0 = utils.deepFreeze(reportFactory.report().build({
+    form: 'report0'
+  }, {
+    patient, submitter: contact0
+  }));
+  const report1 = utils.deepFreeze(reportFactory.report().build({
+    form: 'report1'
+  }, {
+    patient, submitter: contact0
+  }));
+  const report2 = utils.deepFreeze(reportFactory.report().build({
+    form: 'report2'
+  }, {
+    patient, submitter: contact0
+  }));
+  const report3 = utils.deepFreeze(reportFactory.report().build({
+    form: 'report3'
+  }, {
+    patient, submitter: contact0
+  }));
+  const report4 = utils.deepFreeze(reportFactory.report().build({
+    form: 'report4'
+  }, {
+    patient, submitter: contact0
+  }));
+  const report5 = utils.deepFreeze(reportFactory.report().build({
+    form: 'report5'
+  }, {
+    patient, submitter: contact0
+  }));
+
+  const userNoPerms = utils.deepFreeze(userFactory.build({
+    username: 'online-no-perms', place: place1._id, contact: {
+      _id: 'fixture:user:online-no-perms', name: 'Online User',
+    }, roles: [ 'mm-online' ]
+  }));
+  const offlineUser = utils.deepFreeze(userFactory.build({
+    username: 'offline-has-perms', place: place0._id, contact: {
+      _id: 'fixture:user:offline-has-perms', name: 'Offline User',
+    }, roles: [ 'chw' ]
+  }));
+
+  const allDocItems = [ contact0, contact1, contact2, place0, place1, place2, patient ];
+  const allReports = [ report0, report1, report2, report3, report4, report5 ];
+  const allReportsIds = allReports.map(report => report._id);
+  const dataContext = getRemoteDataContext(utils.getOrigin());
+
+  before(async () => {
+    await utils.saveDocs(allDocItems);
+    await utils.saveDocs(allReports);
+    await utils.createUsers([ userNoPerms, offlineUser ]);
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+    await utils.deleteUsers([ userNoPerms, offlineUser ]);
+  });
+
+  describe('GET /api/v1/report/:uuid', async () => {
+    const getReport = Report.v1.get(dataContext);
+
+    it('should return the report matching the provided UUID', async () => {
+      const resReport = await getReport(Qualifier.byUuid(report0._id));
```

Comment:
All these tests should call the api endpoints, instead of request datasource code.